src/models/spaces/cell-zoom/cell-zoom.ts

```diff
@@ -82,8 +83,48 @@ export const kOrganelleInfo: OrganelleInfo = {
 
 export const CellZoomModel = types
   .model("CellZoom", {
-    organisms: types.map(CellMouseModel),
-    rows: types.array(CellZoomRowModel)
+    cellOrganisms: types.optional(types.array(types.maybe(CellMouseModel)),
```

Comment:
This is the piece of state I'd like to discuss. As mice are selected for the cell view, I convert the "backpack mouse" representation into a "cell mouse" representation and store it in this array by "backpack index". If the same mouse is added to a second view, it uses the stored "cell mouse" rather than create a new one. The intention was to simplify substance additions, so that if the substances/color change in one cell view they're reflected in the other. "Cell mice" are only removed from the array when no more views show that backpack mouse.

There were some complications here - I originally wanted to store the "cell mice" in a map, but MST *only* lets you store objects with `id`s by `id`. Storing in an array is fine, except you can't set a value greater than the array length so I needed to initialize it with 6 `undefined` values.

Reply:
It seems slightly cumbersome to me too. Would it work to simply add the backpack reference directly to to the `CellMouseModel`? Then you could just have an array of two `CellMouseModel` that know who their backpack mice are.

When you add a new mouse, you could check the existing `CellMouseModel` to see if it has the same reference.

I may be missing something that makes this not as workable.

Note: In either case (storing a reference or doing what you're doing now) you'll still have index-changing issues when the user deletes any mouse.

Reply:
(I guess GitHub didn't treat this as part of the review because it's a response, so it's out of order. "The index-changing issues" refers to a comment below.)